fighters/koopa/src/status.rs

```diff
@@ -31,4 +54,195 @@ pub unsafe fn exec_special_hi_a(fighter: &mut L2CFighterCommon) -> L2CValue {
     }
     WorkModule::inc_int(fighter.module_accessor, *FIGHTER_KOOPA_STATUS_SPECIAL_HI_WORK_INT_F);
     return 0.into()
+}
+
+// NEUTRAL SPECIAL
+
+#[status_script(agent = "koopa", status = FIGHTER_STATUS_KIND_SPECIAL_N, condition = LUA_SCRIPT_STATUS_FUNC_STATUS_MAIN)]
+unsafe fn special_n_main(fighter: &mut L2CFighterCommon) -> L2CValue {
+    let can_fireball = VarModule::get_int(fighter.battle_object, vars::koopa::instance::FIREBALL_COOLDOWN_FRAME) <= 0;
+    if (!can_fireball){
+        return original!(fighter);
+    }
+    else{
+        fighter.sub_change_motion_by_situation(Hash40::new("special_n_max").into(), Hash40::new("special_air_n_max").into(), false.into());
+        fighter.sub_set_ground_correct_by_situation(true.into());
+
+        fighter.sub_shift_status_main(L2CValue::Ptr(specialnmax_main_loop as *const () as _))
+    }
+}
+
+unsafe extern "C" fn specialnmax_main_loop(fighter: &mut L2CFighterCommon) -> L2CValue {
+    fighter.sub_change_motion_by_situation(Hash40::new("special_n_max").into(), Hash40::new("special_air_n_max").into(), true.into());
+
+    if MotionModule::is_end(fighter.module_accessor) {
+        if StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_AIR {
+            fighter.change_status(FIGHTER_STATUS_KIND_FALL.into(),false.into());
+        }
+        else{
+            fighter.change_status(FIGHTER_STATUS_KIND_WAIT.into(),false.into());
+        }
+        return 0.into();
+    }
+    if fighter.is_situation(*SITUATION_KIND_GROUND){
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_GROUND_STOP);
+        GroundModule::set_correct(fighter.module_accessor, GroundCorrectKind(*GROUND_CORRECT_KIND_GROUND_CLIFF_STOP_ATTACK));
+    }
+    else {
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_AIR_STOP);
+        GroundModule::set_correct(fighter.module_accessor, GroundCorrectKind(*GROUND_CORRECT_KIND_AIR));
+    }
```

Comment:
Should also go behind an `StatusModule::is_situation_changed` check; resetting kinetic type every frame can have strange side effects. This block would also need to go in the pre-loop main func once that is done.

Reply:
Added this to `main`, though if the `pre` status uses ``SITUATION_KIND_NONE, should this block also be in `pre`? Or should this only run when going from ground to air / vice versa (to halt momentum)?``

Reply:
Sorry, I may not have phrased it clearly; you dont need to wrap the block in an `is_situation_changed` check in the pre-main loop function. It's only necessary for the main loop.

So the point is to be changing motion/kinetic type based on situation *once* in `special_n_main` (once you enter the status), then checking for `is_situation_changed` every frame in `special_n_main_loop` (covering a change in situation in the middle of the move).

Reply:
So put the same code block in both main and main_loop, but wrap the block in `is_situation_changed` in main_loop